### PR TITLE
Fix normalization for `DictReplayBuffer`

### LINKED ISSUE #743: [Bug] `DictReplayBuffer` doesn't normalize sampled observations.
### 🐛 Bug

`DictReplayBuffer` doesn't normalize sampled observations : `buffer.sample(batch_size)` and `buffer.sample(batch_size, env)` give exactly the same output.

Probably due to the fact that `env` arg is missing when calling `_normalize_obs` here:
https://github.com/DLR-RM/stable-baselines3/blob/54bcfa4544315fc920be0944fc380fd75e2f7c4a/stable_baselines3/common/buffers.py#L612

Can fix the issue:

```python
obs_ = self._normalize_obs({key: obs[batch_inds, env_indices, :] for key, obs in self.observations.items()}, env)
```

### To Reproduce

```python
from stable_baselines3.common.buffers import DictReplayBuffer
from stable_baselines3.common.env_util import make_vec_env
from stable_baselines3.common.utils import set_random_seed
from stable_baselines3.common.vec_env import VecNormalize

set_random_seed(0)
env = make_vec_env("FetchReach-v1", n_envs=1, seed=0)
env = VecNormalize(env)
buffer = DictReplayBuffer(50, env.observation_space, env.action_space)

obs = env.reset()
for _ in range(50):
    print(obs)
    action = env.action_space.sample()
    next_obs, reward, done, info = env.step(action)
    buffer.add(obs, next_obs, action, reward, done, info)
    obs = next_obs

# buffer.sample(1) and buffer.sample(1, env) give exactly the same output
```

### Expected behavior

Sample should be normalized, thus different from unnormalized sample.

### System Info

 * OS: macOS-12.2-x86_64-i386-64bit Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_X86_64
 * Python: 3.8.12
 * Stable-Baselines3: 1.4.0
 * PyTorch: 1.10.2
 * GPU Enabled: False
 * Numpy: 1.22.1
 * Gym: 0.19.0

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)
- [x] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)

### PR DESCRIPTION
**Pull request by qgallouedec**
## Description

See #743

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

closes #743 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by Miffyli** [comment]
Hey, thanks for the PR! I will take a better look later, but for now: can you add tests to make sure the bug is fixed and to avoid regression in the future? :)

**Comment by qgallouedec** [comment]
Where would you write these tests ? Would you create a `test_buffer.py`? Alternatively, I could include them in `test_normalize.py`, but it doesn't make much sense since it is not the normalization that is tested.

File: tests/test_buffers.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        self._t += 1
+        index = self._t % len(self._observations)
+        obs = {key: self._observations[index] for key in self.observation_space.spaces.keys()}
+        done = self._t >= self._ep_length
+        reward = self._rewards[index]
+        return obs, reward, done, {}
+
+
+@pytest.mark.parametrize("replay_buffer_cls", [ReplayBuffer, DictReplayBuffer])
+def test_normalized_sample(replay_buffer_cls):
```
Comment (Miffyli) [inline tests/test_buffers.py:69]: Nitpick, but maybe call this "test_replay_buffer_normalization" for clarity :). Other than that LGTM!

I will let @araffin take a look at this, too. CI seems to be acting up as well...

**Review by araffin — APPROVED** [review]
Thanks for spotting the issue, LGTM =)

**Pull request merged**